### ISSUE TITLE
[bug] Conan does not provide `Authorization` cookie when accessing `/v1/ping` when adding user 

### ISSUE THREAD

**Issue by timbrown5** [conan-io/conan#6240]
This means that Conan cannot use an Artifactory which is behind a proxy, even if BasicAuth is allowed through the proxy. Also, the log says it is providing an Authorization header, which is misleading.
As we have added a user for this repository it seems logical that the request adds the Authorisation header for that user, and doesn't rely on certain locations being accessible without authorization.

As mentioned in #6066:
Steps:
1) Run local http server on machine using proxy:
`python -m SimpleHTTPServer`
2) Run tcpdump on local machine and use it r print out headers:
`sudo tcpdump -vvv -A -i lo0  port 8000`
3) Run a conan user command and check the headers in the output of `tcpdump` (from 2).
`conan user -p test -r local-repo user`

Doing this we see that `tcpdump` receives these headers:
```
Host: localhost:8000
User-Agent: Conan/1.16.1 (Python 3.7.0) python-requests/2.22.0
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
```
Note there is not `Authorisation` header. Here is the trace log for the came command:
```
{"_action": "COMMAND", "name": "authenticate", "parameters": {"password": "**********", "remote_name": "local-repo"}, "time": 1576150601.185415}
{"_action": "COMMAND", "name": "get_remote_by_name", "parameters": {}, "time": 1576150601.1887949}
{"_action": "REST_API_CALL", "duration": 0.008514165878295898, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.16.1 (Python 3.7.0) python-requests/2.22.0", "X-Client-Anonymous-Id": "**********"}, "method": "GET", "time": 1576150601.1996539, "url": "http://localhost:8000/v1/ping"}
{"_action": "EXCEPTION", "class": "NotFoundException", "message": "Not implemented endpoint. [Remote: local-repo]", "time": 1576150601.2005131}
```
It claims to be adding the `Authorization` header.

To check the method we also ran a curl request (providing a user):
`curl -L -u test:user http://localhost:8000`

The curl request shows it actually sends the `Authorization` header:
```
Host: localhost:8000
Authorization: Basic dGVzdDp1c2Vy
User-Agent: curl/7.54.0
Accept: */*
```

I have tested this using 1.21.0 so this is a bug in the current implementation. This test seems to be testing for this behaviour:
https://github.com/conan-io/conan/blob/81102434601f93b8699fe9f9a482c369ca026c1a/conans/test/functional/remote/auth_bearer_test.py#L69

**Comment by memsharded** [comment]
It is true that the traces are showing some ``"Authorization" = ********`` traces, but the problem lies in the masking of the traces not the request. This request by design never intended to submit basic auth.

I have submitted PR #6245 to do some cleaning on the Rest layer, and also to fix these erroneous traces. We need to investigate further what could be done without breaking other parts (other servers) and other Conan users.

**Comment by timbrown5** [comment]
Could you try requesting without an auth header, then if you get and error
e.g. 403 you could try adding the auth header?
Then the current behaviour is preserved, and the new one only used when
needed?

**Comment by memsharded** [comment]
> Could you try requesting without an auth header, then if you get and error
e.g. 403 you could try adding the auth header?
Then the current behaviour is preserved, and the new one only used when
needed?

Yes, this is one of the lines that we should investigate, in theory it should behave like that, but we need to check better.

**Comment by memsharded** [comment]
Hi @timbrown5 

Are you able to run Conan from sources? We have a exploratory (not guaranteed it will be merged, it is just for research and understanding the issue) pull-request in https://github.com/conan-io/conan/pull/6254. If you can run from sources from that source branch, that would be useful feedback. Thanks!

**Comment by timbrown5** [comment]
Hi @memsharded,

Looks like that fixes the issue. Running the same test as before I get:
```
Host: localhost:8000
User-Agent: Conan/1.22.0-dev (Python 3.6.9) python-requests/2.22.0
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
X-Client-Anonymous-Id: cacc1b7c25b98c191e898a55ec82fe2cf9a18c44
X-Client-Id: 
Authorization: Basic dGVzdDppbnZhbGlk
```

I also tested with Artifactory by turning off our proxy whitelist for `/v1/ping` and `conan user` is added successfully.

**Comment by memsharded** [comment]
Hi @timbrown5 

This has been merged, will be released in Conan 1.22. Meanwhile, as it is in develop is already available in the develop release in test pypi: https://test.pypi.org/

Thanks for the feedback!

**Comment by timbrown5** [comment]
Would it be possible for this to be tagged for the next point release, or is there something restricting it to a minor release only?

**Comment by memsharded** [comment]
Hi @timbrown5 

Conan 1.21.1 was just released, I think there will be no other before 1.22, but lets see. The reason this one was delayed to publish into 1.22.0 is because this seemed not an obvious change, with some risk, and we try that the patch releases do not introduce further risks or breakages. If there is a 1.21.2, I'll have another look.
